### PR TITLE
ClickHouse 23.9 support

### PR DESCRIPTION
**Pull request by Alex-Burmak**

### CODE REVIEW

File: ch_backup/clickhouse/control.py [hunk 1]
```diff
@@ -277,25 +277,14 @@ def __init__(self, ch_ctl_config: dict, main_config: dict) -> None:
...
             "allow_experimental_live_view": 1,
+            "allow_experimental_window_view": 1,
             "allow_suspicious_codecs": 1,
             "allow_suspicious_low_cardinality_types": 1,
         }
-        if self.ch_version_ge("21.9"):
-            settings.update(
-                {
-                    "allow_experimental_database_materialized_mysql": 1,
-                    "allow_experimental_nlp_functions": 1,
```
Comment (Alex-Burmak) [inline ch_backup/clickhouse/control.py:290]: Enabling `allow_experimental_nlp_functions` leads to the following error in 23.9.

```
clickhouse01.test_net_1348 :) set allow_experimental_nlp_functions=1

clickhouse01.test_net_1348 :) SELECT name, create_query FROM system.functions WHERE origin == 'SQLUserDefined'

Received exception from server (version 23.9.1):
Code: 318. DB::Exception: Received from localhost:9000. DB::Exception: No lemmatizers specified in server config on prefix 'lemmatizers'. (INVALID_CONFIG_PARAMETER)
```

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -56,6 +56,8 @@ jobs:
           clickhouse: "22.8.21.38"
         - python: "3.10"
           clickhouse: "23.3.13.6"
+        - python: "3.10"
+          clickhouse: "23.8.3.48"
         - python: "3.10"
           clickhouse: "latest"
```
Comment (Alex-Burmak) [inline .github/workflows/workflow.yml:62]: Explicitly add the LTS version 23.8.

23.9 is `latest` right now. Don't add it explicitly as it has short support lifecycle.

File: ch_backup/clickhouse/control.py [hunk 3]
```diff
@@ -277,25 +277,14 @@ def __init__(self, ch_ctl_config: dict, main_config: dict) -> None:
...
             "allow_experimental_funnel_functions": 1,
             "allow_experimental_live_view": 1,
+            "allow_experimental_window_view": 1,
             "allow_suspicious_codecs": 1,
             "allow_suspicious_low_cardinality_types": 1,
         }
-        if self.ch_version_ge("21.9"):
-            settings.update(
-                {
-                    "allow_experimental_database_materialized_mysql": 1,
```
Comment (Alex-Burmak) [inline ch_backup/clickhouse/control.py:289]: Unconditionally add `allow_experimental_database_materialized_mysql` to `settings` as the oldest supporting version in ch-backup is `22.3`.

File: ch_backup/clickhouse/disks.py [hunk 4]
```diff
@@ -157,12 +157,44 @@ def copy_part(
...
+            ]
+        else:
+            command_args = [
+                "--diskFrom",
+                from_disk,
+                "--diskTo",
+                to_disk,
+                from_path,
+                to_path,
+            ]
```
Comment (Alex-Burmak) [inline ch_backup/clickhouse/disks.py:188]: Options `--diskFrom` and `--diskTo` were renamed to `--disk-from` and `--disk-to` in ClickHouse version 23.9.

**Review by ianton-ru — APPROVED** [review]

**Pull request merged**